src/NServiceBus.Transport.RabbitMQ/Sending/MessageDispatcher.cs

```diff
@@ -38,9 +38,9 @@ public async Task Dispatch(TransportOperations outgoingMessages, TransportTransa
                     tasks.Add(PublishMessage(operation, channel, cancellationToken));
                 }
 
-                channelProvider.ReturnPublishChannel(channel);
-
                 await Task.WhenAll(tasks).ConfigureAwait(false);
+
+                channelProvider.ReturnPublishChannel(channel);
```

Comment:
Would this require now a finally?

Reply:
Yeah, I made a change there to remove the catch/re-throw and instead just have a finally and rely on the pool to handle the dispose as needed.